mev-rs/src/blinded_block_relayer/api/client.rs

```diff
@@ -23,8 +23,20 @@ impl BlindedBlockRelayer for Client {
         self.api.get("/relay/v1/builder/validators").await.map_err(From::from)
     }
 
-    async fn submit_bid(&self, signed_submission: &SignedBidSubmission) -> Result<(), Error> {
-        let response = self.api.http_post("/relay/v1/builder/blocks", signed_submission).await?;
+    async fn submit_bid(
```

Comment:
any reason this is a `String`?

the signature that makes the most sense to me:

```rust
pub async fn submit_bid(&self, signed_submission: &SignedBidSubmission, with_cancellations: bool) -> Result<(), Error>;
```

Reply:
I wanted to address https://github.com/ralexstokes/mev-rs/pull/119#discussion_r1308117262
and decided to go literally off the spec https://flashbots.github.io/relay-specs/#/Builder/submitBlock which specified String. 

I can revert to bool or a u64. Whatever you feel is best.

Reply:
so the *rust* API should be strongly-typed (otherwise why do we bother...)

and I think `bool` is good enough for that for now

then, to address that comment, you'd just match on the incoming data right where you add the query param and convert it correctly (imo this is a weak spot of the API, but let's just follow the specs for now)

so it would like

```rust
if with_cancellations { request = request.query(&[("cancellations", "1")])};
```

Reply:
Thank you for your patience @ralexstokes. Stepped away for an hour and had an "oh duh" moment.